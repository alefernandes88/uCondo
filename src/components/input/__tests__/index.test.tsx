import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import { Input, Select } from '../index'

describe('Components -> Input', () => {
  const labelMock = 'Some label'
  const valueMock = 'Some value'
  const onValueChangeSpy = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render with a normal style', () => {
    const { toJSON } = render((
      <Input
        label={labelMock}
        value={valueMock}
        onValueChange={onValueChangeSpy}
      />
    ))

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with a disabled style', () => {
    const { toJSON } = render((
      <Input
        label={labelMock}
        value={valueMock}
        onValueChange={onValueChangeSpy}
        isDisabled
      />
    ))

    expect(toJSON()).toMatchSnapshot()
  })

  it('should fires callback on text change', async () => {
    const newValueMock = 'new value'
    const { findByTestId } = render((
      <Input
        label={labelMock}
        value={valueMock}
        onValueChange={onValueChangeSpy}
      />
    ))

    const inputField = await findByTestId('input-field')
    fireEvent.changeText(inputField, newValueMock)
    expect(onValueChangeSpy).toBeCalledWith(newValueMock)
  })
})

describe('Components -> Select', () => {
  const labelMock = 'Some label'
  const optionsMock = [
    {
      label: 'Item 1',
      value: 1
    },
    {
      label: 'Item 2',
      value: 2
    }
  ]
  const valueMock = optionsMock[0].value

  const onValueChangeSpy = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render normal style', () => {
    const { toJSON } = render((
      <Select
        label={labelMock}
        value={valueMock}
        onValueChange={onValueChangeSpy}
        options={optionsMock}
      />
    ))

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render disabled style', () => {
    const { toJSON } = render((
      <Select
        label={labelMock}
        value={valueMock}
        onValueChange={onValueChangeSpy}
        options={optionsMock}
        isDisabled
      />
    ))

    expect(toJSON()).toMatchSnapshot()
  })
})
