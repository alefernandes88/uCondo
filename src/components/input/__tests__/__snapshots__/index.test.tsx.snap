// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Input should render with a disabled style 1`] = `
<View
  style={
    {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6A6A6A",
        "fontFamily": "Rubik_500Medium",
        "fontSize": 15,
        "marginBottom": 2,
      }
    }
  >
    Some label
  </Text>
  <View
    isDisabled={true}
    style={
      {
        "backgroundColor": "#d7d7d7",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      }
    }
  >
    <TextInput
      focusable={false}
      onChangeText={[Function]}
      style={
        {
          "color": "#777777",
          "fontSize": 15,
          "paddingBottom": 12,
          "paddingLeft": 17,
          "paddingRight": 17,
          "paddingTop": 12,
        }
      }
      testID="input-field"
      value="Some value"
    />
  </View>
</View>
`;

exports[`Components -> Input should render with a normal style 1`] = `
<View
  style={
    {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6A6A6A",
        "fontFamily": "Rubik_500Medium",
        "fontSize": 15,
        "marginBottom": 2,
      }
    }
  >
    Some label
  </Text>
  <View
    isDisabled={false}
    style={
      {
        "backgroundColor": "#FFF",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      }
    }
  >
    <TextInput
      focusable={true}
      onChangeText={[Function]}
      style={
        {
          "color": "#777777",
          "fontSize": 15,
          "paddingBottom": 12,
          "paddingLeft": 17,
          "paddingRight": 17,
          "paddingTop": 12,
        }
      }
      testID="input-field"
      value="Some value"
    />
  </View>
</View>
`;

exports[`Components -> Select should render disabled style 1`] = `
<View
  style={
    {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6A6A6A",
        "fontFamily": "Rubik_500Medium",
        "fontSize": 15,
        "marginBottom": 2,
      }
    }
  >
    Some label
  </Text>
  <View
    isDisabled={true}
    style={
      {
        "backgroundColor": "#d7d7d7",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      }
    }
  >
    <View>
      <RNCPicker
        items={
          [
            {
              "label": "Item 1",
              "testID": "select-item-1}",
              "textColor": undefined,
              "value": 1,
            },
            {
              "label": "Item 2",
              "testID": "select-item-2}",
              "textColor": undefined,
              "value": 2,
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="select-field"
      />
    </View>
  </View>
</View>
`;

exports[`Components -> Select should render normal style 1`] = `
<View
  style={
    {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6A6A6A",
        "fontFamily": "Rubik_500Medium",
        "fontSize": 15,
        "marginBottom": 2,
      }
    }
  >
    Some label
  </Text>
  <View
    isDisabled={false}
    style={
      {
        "backgroundColor": "#FFF",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      }
    }
  >
    <View>
      <RNCPicker
        items={
          [
            {
              "label": "Item 1",
              "testID": "select-item-1}",
              "textColor": undefined,
              "value": 1,
            },
            {
              "label": "Item 2",
              "testID": "select-item-2}",
              "textColor": undefined,
              "value": 2,
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={0}
        style={
          [
            {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="select-field"
      />
    </View>
  </View>
</View>
`;
